File: spec/python/test_nav_parent_switch.py
import unittest

from nav_parent_switch import NavParentSwitch

class TestNavParentSwitch(unittest.TestCase):
    def test_nav_parent_switch(self):
        r = NavParentSwitch.from_file("src/nav_parent_switch.bin")

        self.assertEqual(r.category, 1)
        self.assertEqual(r.content.foo, 0x42)
        self.assertEqual(r.content.subelement.bar, 0xff)

File: spec/lua/test_nav_parent_switch.lua
local luaunit = require("luaunit")

require("nav_parent_switch")

TestNavParentSwitch = {}

function TestNavParentSwitch:test_nav_parent_switch()
    local r = NavParentSwitch:from_file("src/nav_parent_switch.bin")

    luaunit.assertEquals(r.category, 1)
    luaunit.assertEquals(r.content.foo, 0x42)
    luaunit.assertEquals(r.content.subelement.bar, 0xff)
end
